cmd/plume/prune.go

```diff
@@ -0,0 +1,192 @@
+	cmdPrune.Flags().StringVar(&azureTestContainer, "azure-test-container", "", "Use test container instead of default")
+	cmdPrune.Flags().BoolVarP(&pruneDryRun, "dry-run", "n", false,
+		"perform a trial run, do not make changes")
+	AddSpecFlags(cmdPrune.Flags())
+	root.AddCommand(cmdPrune)
+}
+
+func runPrune(cmd *cobra.Command, args []string) {
+	if len(args) > 0 {
+		plog.Fatal("No args accepted")
+	}
+
+	spec := ChannelSpec()
+	ctx := context.Background()
+	pruneAWS(ctx, &spec)
+	pruneAzure(ctx, &spec)
+}
+
+func pruneAzure(ctx context.Context, spec *channelSpec) {
+	if spec.Azure.StorageAccount == "" || azureProfile == "" {
+		plog.Notice("Azure image pruning disabled, skipping.")
+		return
+	}
+
+	for _, environment := range spec.Azure.Environments {
+		api, err := azure.New(&azure.Options{
+			AzureProfile:      azureProfile,
+			AzureAuthLocation: azureAuth,
+			AzureSubscription: environment.SubscriptionName,
+		})
+		if err != nil {
+			plog.Fatalf("Failed to create Azure API: %v", err)
+		}
+		if err := api.SetupClients(); err != nil {
+			plog.Fatalf("Failed to set up clients: %v", err)
+		}
+
+		plog.Printf("Fetching Azure storage credentials for %q in %q", spec.Azure.StorageAccount, spec.Azure.ResourceGroup)
+
+		storageKey, err := api.GetStorageServiceKeysARM(spec.Azure.StorageAccount, spec.Azure.ResourceGroup)
+		if err != nil {
+			plog.Fatalf("Failed to fetch storage key: %v", err)
+		}
+		if storageKey.Keys == nil {
+			plog.Fatalf("No storage service keys found")
+		}
+
+		container := spec.Azure.Container
+		if azureTestContainer != "" {
+			container = azureTestContainer
+		}
+
+		for _, key := range *storageKey.Keys {
+			blobs, err := api.ListBlobs(spec.Azure.StorageAccount, *key.Value, container, storage.ListBlobsParameters{})
+			if err != nil {
+				plog.Warningf("Error listing blobs: %v", err)
+			}
+			plog.Infof("Got %d blobs for container %q (key %v)\n", len(blobs), container, key)
+
+			now := time.Now()
+			for _, blob := range blobs {
+				lastModifiedDate, err := time.Parse(time.RFC1123, blob.Properties.LastModified)
+				if err != nil {
+					plog.Warningf("Error converting last modified date (%v): %v", blob.Properties.LastModified, err)
+				}
+				duration := now.Sub(lastModifiedDate)
+				daysOld := int(duration.Hours() / 24)
+				if daysOld >= days {
+					plog.Infof("Old blob %q: %d days old\n", blob.Name, daysOld)
+					if !pruneDryRun {
+						plog.Infof("Deleting Blob %q in container %q", blob.Name, container)
+						err = api.DeleteBlob(spec.Azure.StorageAccount, *key.Value, container, blob.Name)
+						if err != nil {
+							plog.Warningf("Error deleting blob (%v): %v", blob.Name, err)
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+func pruneAWS(ctx context.Context, spec *channelSpec) {
+	if spec.AWS.Image == "" || awsCredentialsFile == "" {
+		plog.Notice("AWS image pruning disabled.")
+		return
+	}
+
+	for _, part := range spec.AWS.Partitions {
```

Comment:
I think it would be better to be able to specify one partition instead of iterating over all here.

Reply:
Or maybe I'm missing that the channel and partition is passed as argument

Reply:
Ah, ok, found it ;)

Reply:
I've added a comment to clarify this. Thanks.